src/init.cpp

```diff
@@ -1167,6 +1167,20 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
     assert(!node.addrman);
     auto check_addrman = std::clamp<int32_t>(args.GetArg("-checkaddrman", DEFAULT_ADDRMAN_CONSISTENCY_CHECKS), 0, 1000000);
     node.addrman = std::make_unique<CAddrMan>(/* deterministic */ false, /* consistency_check_ratio */ check_addrman);
+    {
```

Comment:
What is the reasoning for moving this to init.cpp? I think this goes in the wrong direction, of having parts take care of their own initialization instead of init.cpp being a hodge-podge of initialization functionality.

Reply:
Since #20228, addrman is owned by the `node.context` struct and gets constructed inside the `AppInitMain()` function. That PR includes motivation for why that is.

Since `AppInitMain()` is responsible for constructing the addrman, it should also be responsible for making sure addrman deserializes from peers.dat (and recovers if there's an error). Before this PR, addrman peers.dat derserialization is happening in `CConnMan::Start()`, which doesn't make a lot of sense.

Reply:
Ok, I agree. I hope it can be factored out again at some point to a place that makes sense.

Reply:
See #22754